android/libraries/rib-base/src/main/kotlin/com/uber/rib/core/BasicInteractor.kt

```diff
@@ -13,15 +13,14 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.uber.rib.core.lifecycle;
-
-import com.uber.rib.core.Worker;
+package com.uber.rib.core
 
 /**
- * Enum used represent when a {@link Worker} should stop and start in response to interactor
- * lifecycle events.
+ * [Interactor] that doesn't rely on field injection.
+ *
+ * @param <P> the type of [Presenter].
+ * @param <R> the type of [Router].
  */
-public enum WorkerEvent {
-  START,
-  STOP
-}
+abstract class BasicInteractor<P : Any, R : Router<*>> protected constructor(
```

Comment:
P: Any or P : Presenter

Reply:
`P : Any` right now since in Java it is currently unbounded